### PR TITLE
Support for AIX

### PR DESCRIPTION
**Pull request by AkashKTripathi**
We are using this Issue to push the Code changes that are needed to provide IBM AIX OS support for the cloud-init.

In multiple location, where the command are specific to IBM AIX, We have inserted the if condition to check the underlying OS.
We are running AIX specific command only if the AIX is running on that system.

### CODE REVIEW

File: cloudinit/sources/DataSourceNoCloud.py [hunk 1]
```diff
@@ -37,15 +39,17 @@ def __str__(self):
...
-        fslist.extend(util.find_devs_with("TYPE=iso9660"))
-
-        label_list = util.find_devs_with("LABEL=%s" % label.upper())
-        label_list.extend(util.find_devs_with("LABEL=%s" % label.lower()))
-        label_list.extend(util.find_devs_with("LABEL_FATBOOT=%s" % label))
-
-        devlist = list(set(fslist) & set(label_list))
-        devlist.sort(reverse=True)
+        if platform.system().lower() == "aix":
+            devlist = aix_util.find_devs_with("cd0")
```
Comment (igalic) [inline cloudinit/sources/DataSourceNoCloud.py:43]: i know the structure in util.py isn't great, but, why aren't you extending that, rather tumač touching every DataSource separately?

File: config/cloud.cfg [hunk 2]
```diff
@@ -0,0 +1,102 @@
+# The top level settings are used as module
```
Comment (igalic) [inline config/cloud.cfg:1]: this file needs to go

File: sysvinit/aix/cloud-config [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+
+RETVAL=0
+
+prog="cloud-init"
+cloud_init="/opt/freeware/bin/cloud-init"
+conf="/opt/freeware/etc/cloud/cloud.cfg"
+
+# If there exist sysconfig/default variable override files use it...
+[ -f /etc/sysconfig/cloud-init ] && . /etc/sysconfig/cloud-init
+[ -f /etc/default/cloud-init ] && . /etc/default/cloud-init
```
Comment (igalic) [inline sysvinit/aix/cloud-config:50]: other init systems also allow disabling with the existence of a file

**Review by holmanb — CHANGES_REQUESTED** [review]
Hey @AkashKTripathi,

Thanks for this contribution to cloud-init. As described in the [contribution guide](https://cloudinit.readthedocs.io/en/latest/development/index.html#prerequisites), I'll need you to sign the CLA before this contribution can be accepted.

File: cloudinit/cmd/main.py [hunk 4]
```diff
@@ -386,7 +386,12 @@ def main_init(name, args):
                 LOG.debug("manual cache clean found from marker: %s", mfile)
                 existing = "trust"
 
-        init.purge_cache()
+        # rm_instance_lnk is now set to True. Else, cache will be used in case
+        # instead of optical disk to fetch network config. This check make sure 
+        # that customer always uses latest config data even if the instance-id 
+        # is matching
+        init.purge_cache(True)
```
Comment (holmanb) [inline cloudinit/cmd/main.py:393]: One more comment:

This is a pretty big change to cloud-init. What justification do you have for it? Also, which datasources have you tested this on? Wouldn't this break any user using [nocloud](https://cloudinit.readthedocs.io/en/latest/reference/datasources/nocloud.html#method-1-local-filesystem-labeled-filesystem) with only a `CIDATA` drive?

**Comment by holmanb** [comment]
@AkashKTripathi Any update on this?

**Comment by holmanb** [comment]
@AkashKTripathi Is there any interest in making this happen still?

**Comment by AkashKTripathi** [comment]
> @AkashKTripathi Is there any interest in making this happen still?

Hi @holmanb
we are working on it.  we will update you

**Pull request closed without merging**